### PR TITLE
Add Docker E2E environment setup and runner

### LINKED ISSUE #2576: Refactor E2E Setup Script to Focus Solely on Setup
**Is your feature request related to a problem? Please describe.**
Currently, the setup script for E2E testing is responsible for both setting up the environment and running the tests. This can make it difficult to manage and maintain the testing workflow. To improve this, we should refactor the setup script to focus only on setting up the environment, while the existing test execution script handles running the tests.

**Describe the solution you'd like**
- Update the setup script to handle only the environment setup, including the necessary installations and configurations, and remove any test execution code from the script.

**Describe alternatives you've considered**
N/A

**Additional context**
https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2522#issuecomment-1473224875

### PR DESCRIPTION
**Pull request by diegocurbelo**
Fixes #2576, #2530

## Changes proposed in this Pull Request:
This PR adds an E2E test workflow for open PRs.

The `npm run test:e2e-setup` and `npm run test:e2e` scripts were extended to support not receiving a `base_url` parameter.

When no `base_url` param is present, an independent E2E docker environment is configured/used for the tests, which allows the tests to run in a GitHub Workflow on each Pull Request.

### Verify that **all** PR checks pass
Including the new `E2E tests / WP=latest, WC=latest, PHP=7.4 (pull_request)`:

Screenshot 2024-02-21 at 12 44 27

### Test the Jurassic Ninja runner:
1. Create a JN site
2. Update the credentials for the newly created JN site in your `tests/e2e/config/local.env`
3. Run the E2E suite with the JN base url:
    `npm run test:e2e-setup -- --base_url=<JN_URL>`
4. Verify that the setup and all 22 e2e tests succeed
    Screenshot 2024-02-21 at 12 43 42

### Test the docker runner:
0. Build the extensions asses
    `npm run build`

1. Initialize/start the E2E docker containers:
    `npm run test:e2e-setup`

2. Verify that the environment was successfully started: 
    Screenshot 2024-02-21 at 12 32 29

3. Run the E2E suite (without the `base_url` param):
    `npm run test:e2e`

6. Verify that the setup and all 22 e2e tests succeed:
    Screenshot 2024-02-21 at 12 35 04

7. Stop the E2E docker containers:
    `npm run test:e2e-down`

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+on:
+  pull_request
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
Comment (wjrosa) [inline .github/workflows/e2e-tests.yml:12]: We may consider alpine later in case this gets slow 👀

**Review by wjrosa — APPROVED** [review]
Amazing work here, Diego! It works very well. As I mentioned in Slack, I have a test that is failing (`refunds` property does not exist), and had an issue trying to run the local site container (very edge case). 

Since those are not directly related to your changes, I think it is good to go 👍
